sequential.py

```diff
@@ -0,0 +1,160 @@
+            
+            if old_log_file_name == new_log_file_name:
+                one_list.append(record)
+                if new_record_id != old_record_id+1:
+                    record_id = new_record_id
+                    validation_result.append(ValidationResult(False, "Detected incorrectly incremented recordId. Record Number: '%d' Expected '%d' but got '%d'" % (record_num, old_record_id+1, new_record_id), record))
+                if new_serial_number != old_serial_number+1:
+                    record_id = new_record_id
+                    validation_result.append(ValidationResult(False, "Detected incorrectly incremented serialNumber. Record Number: '%d' Expected '%d' but got '%d'" % (record_num, old_serial_number+1, new_serial_number), record))
+                if new_record_generated_at < old_record_generated_at:
+                    record_id = new_record_id
+                    validation_result.append(ValidationResult(False, "Detected non-chronological recordGeneratedAt. Record Number: '%d' Previous timestamp was '%s' but current timestamp is '%s'" % (record_num, old_record_generated_at, new_record_generated_at), record))
+                if new_ode_received_at < old_ode_received_at:
+                    record_id = new_record_id
+                    validation_result.append(ValidationResult(False, "Detected non-chronological odeReceivedAt. Record Number: '%d' Previous timestamp was '%s' but current timestamp is '%s'" % (record_num, old_ode_received_at, new_ode_received_at), record))
+            else:
+                #validation_result.append(ValidationResult(True, "New log file detected. Resetting old item values. Record Number: '%d' Old filename: '%s', new filename: '%s'" % (record_num, old_log_file_name, new_log_file_name)))
+                bundle_size_validation_result = self.validate_bundle_size(one_list)
+                if bundle_size_validation_result.__len__() > 0:
+                    validation_result.extend(bundle_size_validation_result)
+                    record_id = new_record_id
+                    break
+                one_list.clear()
+                one_list.append(record)
+
+            old_log_file_name = new_log_file_name
+            old_record_id = new_record_id
+            old_serial_number = new_serial_number
+            old_record_generated_at = new_record_generated_at
+            old_ode_received_at = new_ode_received_at
+
+            if record_id == 0:
+                break
+            
+        return validation_result
+
+    def validate_bundle_size(self, sorted_record_list):
+        record_id = 0
+        first_record_id = int(sorted_record_list[0]['metadata']['serialId']['recordId'])
+        last_record_id = int(sorted_record_list[-1]['metadata']['serialId']['recordId'])
+
+        validation_result = []
+        # partial or full list?
+        if first_record_id == 1:
+            # head of a partial list?
+            if last_record_id == first_record_id:
```

Comment:
So this handles a full list. Are we ignoring partial lists? We could still implement the check that `max_record_id < bundleSize - 1`

Reply:
The else part of `if first_record_id == 1` handles partial `tail` list. There is no check of `bundleSize` for a a partial `head` list because there is nothing there to test.